addons/sale_loyalty/static/src/components/loyalty_card/loyalty_card.js

```diff
@@ -0,0 +1,32 @@
+import { standardFieldProps } from '@web/views/fields/standard_field_props';
+import { registry } from '@web/core/registry';
+import {
+    Component,
+    onWillRender,
+    toRaw,
+} from "@odoo/owl";
+
+export class LoyaltyCardComponent extends Component {
+    static template = "sale_loyalty.sale_order_loyalty_table";
+    static props = {
+        ...standardFieldProps,
+    };
+
+    setup() {
+        this.loyalty_card = {};
+        this.formatData(this.props);
+        onWillRender(() => this.formatData(this.props));
+    }
+
+    formatData(props) {
+        const loyalty_card_values = toRaw(props.record.data[this.props.name]);
+        this.props.loyalty_card = loyalty_card_values["loyalty_card"]  || undefined;
+        this.props.point_name = loyalty_card_values['point_name']
+    }
+}
+
+export const loyaltyCardComponent = {
+    component: LoyaltyCardComponent,
+};
+
+registry.category("fields").add("loyalty-card-values", loyaltyCardComponent);
```

Comment:
Why this is in field category :thinking: 
You should create view_widget as this is not related to specific field IMO